pkg/image-manager/controller.go

```diff
@@ -113,12 +151,22 @@ func (h *imageEntityHandler) Add(ctx context.Context, obj entitystore.Entity) (e
 		log.Error(err)
 	}
 
+	if h.CheckAndLock(i) {
+		return nil
+	}
+
+	defer h.Unlock(i)
 	defer func() { h.Store.UpdateWithError(ctx, i, err) }()
 
+	log.Infof("Creating image %s/%s", i.OrganizationID, i.Name)
 	if err = h.Builder.imageCreate(ctx, i, &bi); err != nil {
+		i.Status = entitystore.StatusERROR
```

Comment:
`UpdateWithError` already checks if err is not nil and sets the status accordingly, so this seems redundant.